### DISCUSSION TITLE
Feedback: Comments on individual commits (made outside a pull request) no longer appear in the pull request timeline

### DISCUSSION THREAD

**Discussion by willsmythe** [community/community#28924] [Pull Requests]
We recently rolled out a change ([changelog](https://github.blog/changelog/2022-08-04-commit-comments-no-longer-appear-in-the-pull-request-timeline/)) to no longer show comments added to individual commits (comments added outside the context of a pull request) on the pull request timeline. Previously, if a pull request happened to include a commit that had commit comments, those comments would appear on that pull request's timeline. Users often reported that these comments were irrelevant to the pull request or caused confusion when added by someone not involved in the review.

### What changed

Prior to this change, comments on individual commits (made outside the context of the pull request) would appear in the timeline of any pull request that included that commit:

<img src="https://user-images.githubusercontent.com/2503052/183735557-33c33c83-ecff-4365-aaba-4e413686ada7.png" width="700px">

With this change, commit comments (like the highlighted `029541b` comment) no longer appear in the pull request timeline.

The timeline events [REST](https://docs.github.com/en/rest/issues/timeline) and [GraphQL](https://docs.github.com/en/graphql/reference/objects#pullrequest) APIs also no longer return commit comments associated with any commits included in the pull request.

### What did **not** change

This change did not impact a user's ability to add, view, or manage comments on individual commits.

* Users can still add comments to individual commits or changed lines in a commit
* Users can still review pull requests one commit at a time and add review comments
* Commit comments still surface on the commits listing page
* Commit comments still surface at the bottom of the commit page
* Commit comments are still accessible from the **Commits** tab of the pull request page
* Commits pushed to a pull request head branch still surface in the timeline and on the **Commits** tab of the pull request page

There was also no impact to **_commit messages_**, i.e. the message stored with the Git commit. Commit messages continue to surface everywhere they previously surfaced, including on the pull request page.

Let us know what you like or don't like about this change.

**Comment by tobiasviehweger** [comment]
I've got a question regarding this, because I'm not sure if this is the same thing. Yesterday, I started with the review of a pull request, but did not completely finish with all the files. To allow the PR opener to work on the partial comments, I submitted the review with "Comment". Today, when using the "Files changed" tab, to continue working on the PR, my comments from yesterday are not shown inline anymore, but only on the main commit UI - is this a side effect of that change?

![image](https://user-images.githubusercontent.com/2111803/183835810-1b6544c1-c379-454f-857c-ee51c4e018e1.png)
![image](https://user-images.githubusercontent.com/2111803/183835912-5499922d-9237-48df-94d1-642914745a04.png)

**Reply by willsmythe** [reply]
This is unrelated to the announced change. 

If you navigate to the Conversation tab of the pull request, do you see your comments marked as "outdated"? This can happen if changes were pushed to the file after you commented. If this is the case and you click the link on the comment, you should get taken to the Files Changed (or Commits) tab filtered to a specific commit and should see your comment again.

If that doesn't work, I would contact [GitHub Support](https://support.github.com/) since they'll have access to more details and can help figure out what's going on!

**Comment by donatj** [comment]
So we attach human readable diffs to commits with lock file changes as they can be quite impenetrable otherwise.

here is a simple example 

<img width="1164" alt="image" src="https://user-images.githubusercontent.com/133747/184405071-bec554ba-1a99-4f32-8f07-deb11cdb6f7d.png">

These diffs no longer show on Pull Requests, and that's a problem for us.

**Comment by frank-segui** [comment]
Hey there, we have a case were we build a preview site and post the url in the workflow to the commit. Then users were able to see the comment on the pr and know where to go. With this change that comment is no longer obvious. The workflow can run before the PR is created so adding it to the pr is not that feasible. Is there a way to handle this case after the change?

**Comment by devinrhode2** [comment]
Hmm... So if someone comments on a certain line of a commit, and then PR author force pushes new commits (trying to keep clean history)... those comments are basically _lost_. The notifications feed on github.com might not even show what's going on there either, there can be other comments at the bottom of the actual Pull Request page, and, I guess they would have no idea there's _also_ some comments on a commit that are basically lost now.

I guess, something to check on: Do comments on commits create a new "`<li>`" on the /notifications page? I suppose the notifications page could get too messy that way. If you frequently force push new commit history, each unique sha basically makes an appearance on the /notifications page... there could be a lot of "commit comment threads" that "clutter"(?) up the /notifications page.

**Comment by gre** [comment]
we have CI automation that is running on commits and gives a nice report by commenting on the commit. The fact we no longer see the report also on the PR is a regression to us and we would love to have the ability to have it back, maybe as a settings?

**Reply by pefribeiro** [reply]
Very odd for this functionality to just suddenly disappear. I had to Google for this change as it all went under the radar.

**Comment by deitry** [comment]
Is there a single place to see all the commit comments in the repo? It feels more and more difficult to get into conversation if someone makes a comment on a commit that you was not involved with and was not directly mentioned.

I think it's a big omission that there is no separate "commit comments" category in repo's Discussions page.

**Comment by eroller** [comment]
It would be nice if, when commenting on a commit, we could have the option to add the comment to the associated PR rather than to the commit (or maybe both?). I don't know how this would interact with force pushes on the PR, but it is definitely nice seeing the comments and diffs within the PR directly and being able to make sure each one is resolved before approving the PR.

**Comment by revolter** [comment]
So, is it true that there is no way to review a PR by asking the OP to change something with an entire commit?

**Reply by deitry** [reply]
There is an option to "request changes"
<img width="499" alt="image" src="https://github.com/community/community/assets/6714840/d868c6c5-1cfa-42a3-8a1a-8d1859aaf4f3">

**Reply by revolter** [reply]
Both this discussion and my comment refer to comments on commits.

**Reply by deitry** [reply]
Yes they are, but you said "there is no way .." when there _is_ actually another way - and it works better (well, in terms of GitHub UI) when you want to point out that changes should be made.

**Reply by revolter** [reply]
You didn't understand me. It looks like you can link a review comment to a line of code, but not to an entire commit.

**Comment by floooh** [comment]
Can  we disable this 'comment-on-commits' feature alltogether somehow? I'm getting notifications like this from time to time:

<img width="880" alt="Screenshot 2025-01-13 at 12 04 05" src="https://github.com/user-attachments/assets/f2ddf09b-62df-4f76-9214-5264e6519e68" />

...and clicking the notification just goes to the entire commit, but not the actual comment. So I need to scroll pages after pages to eyeball-scan for a little icon on the right side which hides the comment. This is **really** bad UX, and I'd rather disable this feature entirely across my repositories and instead encourage users to write regular issues.

<img width="767" alt="Screenshot 2025-01-13 at 12 05 26" src="https://github.com/user-attachments/assets/06929cb4-50aa-4eca-ad18-c2674bf41611" />

**Comment by ArtemLukanin-TomTom** [comment]
I receive notifications about the new commits, click on it and comment, what should be changed... but the comments are not visible in the PR, which I reviewed! Do you expect people to review the whole PR over and over with every commit and compare the commit view and the PR view to find the place, where to put the comment? At least fix the email notification and remove the link to the commit to avoid this trap.
